Allow libxc to build against uclibc, which has no fadvise

diff -r 7f4492d15359 tools/libxc/xc_linux.c
--- a/tools/libxc/xc_linux.c	Mon Jan 24 12:37:12 2011 +0000
+++ b/tools/libxc/xc_linux.c	Mon Jan 24 12:37:49 2011 +0000
@@ -44,12 +44,14 @@ void discard_file_cache(xc_interface *xc
         cur &= ~(XC_PAGE_SIZE-1);
     }
 
+#ifndef __UCLIBC__
     /* Discard from the buffer cache. */
     if ( posix_fadvise64(fd, 0, cur, POSIX_FADV_DONTNEED) < 0 )
     {
         /*PERROR("Failed to discard cache: %s", strerror(errno));*/
         goto out;
     }
+#endif
 
  out:
     errno = saved_errno;
diff -r 7f4492d15359 tools/libxc/xc_private.c
--- a/tools/libxc/xc_private.c	Mon Jan 24 12:37:12 2011 +0000
+++ b/tools/libxc/xc_private.c	Mon Jan 24 12:37:49 2011 +0000
@@ -49,7 +49,7 @@
 static int xc_osdep_get_info(xc_interface *xch, xc_osdep_info_t *info)
 {
     int rc = -1;
-#ifndef __MINIOS__
+#if !defined(__MINIOS__) && !defined(__UCLIBC__)
     const char *lib = getenv(XENCTRL_OSDEP);
     xc_osdep_info_t *pinfo;
     void *dl_handle = NULL;
@@ -93,7 +93,7 @@ static int xc_osdep_get_info(xc_interfac
 
     rc = 0;
 
-#ifndef __MINIOS__
+#if !defined(__MINIOS__) && !defined(__UCLIBC__)
 out:
     if ( dl_handle && rc == -1 )
         dlclose(dl_handle);
@@ -104,7 +104,7 @@ out:
 
 static void xc_osdep_put(xc_osdep_info_t *info)
 {
-#ifndef __MINIOS__
+#if !defined(__MINIOS__) && !defined(__UCLIBC__)
     if ( info->dl_handle )
         dlclose(info->dl_handle);
 #endif
